client/relay-chain-interface/src/lib.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::sync::Arc;
+
+use cumulus_primitives_core::{
+	relay_chain::{
+		v1::{CommittedCandidateReceipt, OccupiedCoreAssumption, SessionIndex, ValidatorId},
+		Block as PBlock, BlockId, Hash as PHash, InboundHrmpMessage,
+	},
+	InboundDownwardMessage, ParaId, PersistedValidationData,
+};
+use polkadot_overseer::Handle as OverseerHandle;
+use sc_client_api::{blockchain::BlockStatus, StorageProof};
+
+use sp_api::ApiError;
+use sp_core::sp_std::collections::btree_map::BTreeMap;
+use sp_state_machine::StorageValue;
+
+pub enum BlockCheckResult {
+	InChain,
+	NotFound(sc_client_api::ImportNotifications<PBlock>),
+}
+
+/// Should be used for all interaction with the relay chain in cumulus.
+pub trait RelayChainInterface: Send + Sync {
+	fn get_storage_by_key(&self, block_id: &BlockId, key: &[u8]) -> Option<StorageValue>;
```

Comment:
Docs?

Also weird that we return `None` here for when the key doesn't exist or there was an error.